### PR TITLE
Update language server extension guide to latest version of the code

### PR DESCRIPTION
**Pull request by tomaszbartoszewski**
All changes are updates to show latest version of the code, you can see it in the vscode-extension-samples repo:

https://github.com/microsoft/vscode-extension-samples/blob/83261e3f32421da84513c5a76e8b6a08fff3c332/lsp-sample/client/package.json#L12

https://github.com/microsoft/vscode-extension-samples/blob/83261e3f32421da84513c5a76e8b6a08fff3c332/lsp-sample/server/package.json#L14

https://github.com/microsoft/vscode-extension-samples/blob/master/lsp-sample/server/src/server.ts

https://github.com/microsoft/vscode-extension-samples/blob/83261e3f32421da84513c5a76e8b6a08fff3c332/lsp-sample/.vscode/launch.json#L25

https://github.com/microsoft/vscode-extension-samples/blob/83261e3f32421da84513c5a76e8b6a08fff3c332/lsp-sample/client/src/test/completion.test.ts#L10

https://github.com/microsoft/vscode-extension-samples/blob/83261e3f32421da84513c5a76e8b6a08fff3c332/lsp-sample/client/src/test/helper.ts#L19

### CODE REVIEW

File: api/language-extensions/language-server-extension-guide.md [hunk 1]
```diff
@@ -459,52 +455,53 @@ To add document validation to the server, we add a listener to the text document
...
+  // The validator creates diagnostics for all uppercase words length 2 and more
+  let text = textDocument.getText();
+  let pattern = /\b[A-Z]{2,}\b/g;
+  let m: RegExpExecArray | null;
+
+  let problems = 0;
+  let diagnostics: Diagnostic[] = [];
+  while ((m = pattern.exec(text)) && problems < settings.maxNumberOfProblems) {
+    problems++;
+    let diagnostic: Diagnostic = {
```
Comment (RMacfarlane) [inline api/language-extensions/language-server-extension-guide.md:472]: end of document test

File: api/language-extensions/language-server-extension-guide.md [hunk 2]
```diff
@@ -379,13 +395,13 @@ async function validateTextDocument(textDocument: TextDocument): Promise<void> {
     diagnostics.push(diagnostic);
   }
 
-  // Send the computed diagnostics to VS Code.
+  // Send the computed diagnostics to VSCode.
```
Comment (gregvanl) [inline api/language-extensions/language-server-extension-guide.md:398]: We like to use "VS Code" with a space, so it was updated in the extension guide. We should update the sample to use he same name.

File: api/language-extensions/language-server-extension-guide.md [hunk 3]
```diff
@@ -379,13 +395,13 @@ async function validateTextDocument(textDocument: TextDocument): Promise<void> {
...
   connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
 }
 
 connection.onDidChangeWatchedFiles(_change => {
-  // Monitored files have change in VS Code
-  connection.console.log('We received a file change event');
+  // Monitored files have change in VSCode
```
Comment (gregvanl) [inline api/language-extensions/language-server-extension-guide.md:403]: "VS Code" with a space.
We should also update the sample to use the correct grammar "a file change event".

File: api/language-extensions/language-server-extension-guide.md [hunk 4]
```diff
@@ -459,52 +455,53 @@ To add document validation to the server, we add a listener to the text document
...
-                {
-                    location: {
-                        uri: textDocument.uri,
-                        range: Object.assign({}, diagnostic.range)
-                    },
-                    message: 'Particularly for names'
-                }
-            ];
+documents.onDidChangeContent(async(change) => {
+  let textDocument = change.document
```
Comment (gregvanl) [inline api/language-extensions/language-server-extension-guide.md:459]: This looks like a necessary addition to get the textDocument. Can you add a trailing semi-colon to the statement.

File: api/language-extensions/language-server-extension-guide.md [hunk 5]
```diff
@@ -459,52 +455,53 @@ To add document validation to the server, we add a listener to the text document
...
-        diagnostics.push(diagnostic);
+      ];
     }
+    diagnostics.push(diagnostic);
+  }
 
-    // Send the computed diagnostics to VS Code.
-    connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
-}
+  // Send the computed diagnostics to VSCode.
```
Comment (gregvanl) [inline api/language-extensions/language-server-extension-guide.md:502]: "VS Code"

**Comment by gregvanl** [comment]
Thanks for doing this. I'll make the couple of changes mentioned above.

**Comment by tomaszbartoszewski** [comment]
Thank you, I'm sorry I didn't have time yesterday to correct it.

**Comment by gregvanl** [comment]
No worries, just wanted to get your changes in and published. Thanks again!

**Pull request merged**